pkg/utils/spinner.go

```diff
@@ -1,21 +1,35 @@
 package utils
 
 import (
+	"context"
 	"github.com/briandowns/spinner"
+	"os"
+	"os/signal"
 	"time"
 )
 
 func NewSpinner(message string, maxRetries int, function func() error) error {
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I am fairly certain there is a shortcut to this.

```
ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
defer stop()
```

Reply:
That said, I do think there are some dragons in registering signal handlers here. May be well enough for v1, since it does solve the problem. Just want to bring up the dangers in case you judge it not worth it.

Those dragons: it is probably best practice to leave signal handling to `main` (or somewhere obvious and close to main.) Registering and un-registerting signal handlers is a global thing. Were there multiple things happening (a nonsense example would be two spinners on stdout and stderr) one of them would "win". Or if we installed signal handlers in main, I believe this would end up stomping those.